### DISCUSSION TITLE
Question about entities before proceeding.

### DISCUSSION THREAD

**Discussion by alvinchen1** [emporia-vue-local/esphome#151] [Q&A]
Hi! Been using this custom component for a while.  https://github.com/magico13/ha-emporia-vue.  Was interested in going cloud-less/local only and found this procedure.  I have very early understanding of ESPHome and read through the yaml template.  The custom component (ha-emporia-vue) gives me 3 entities for each circuit.  1 min (unit of measurement Watts), 1 day (kWh) and 1 month (kWh). 

<img width="774" alt="image" src="https://user-images.githubusercontent.com/26409535/230775385-90e38a79-3e24-4e1e-a39c-04fee42e7606.png">

I use all three in various automations and was wondering if I did this ESPHome flash, would I get all three?  From the looks of it, I see one which is named 'Total power' which is in Watts which I assume equivalent to the 1 min entity, and one called "Total energy" which I think is equivalent for the 1 day entity.  It doesn't look like there is one for 1 month, but I think I could create one with a HomeAssistant Utility Meter Template.  

I read this thread https://github.com/emporia-vue-local/esphome/discussions/117 and from the screenshot there, I think that's what I'm seeing. 

Was just hoping someone could confirm before I moved forward.  Thanks so much!

**Comment by flaviut** [comment]
> Total power' which is in Watts which I assume equivalent to the 1 min entity

not quite, it is the actual instantaneous power draw at that point in time & updates every few seconds. It updating so often allows for automation & is one of the big advantages of this integration over the cloud integration.

> "Total energy" which I think is equivalent for the 1 day entity.

You can use it like that, sure. It will reset if the vue is restarted or the power goes out.

>  It doesn't look like there is one for 1 month, but I think I could create one with a HomeAssistant Utility Meter Template.

I've never seen that feature before, but that looks cool & sounds like what you want!
